main.tf

```diff
@@ -131,6 +131,22 @@ resource "aws_instance" "di_web_linux" {
   subnet_id                   = "${aws_subnet.data_ingest.id}"
   private_ip                  = "${var.dp_web_linux_private_ip}"
 
+  user_data = <<EOF
+#!/bin/bash
+
+if [ ! -f /bin/aws ]; then
+    curl https://bootstrap.pypa.io/get-pip.py | python
+    pip install awscli
+
+SFTP_USERNAME=`aws --region eu-west-2 ssm get-parameter --name mock_sftp_ftp_server_sftp_username --query 'Parameter.Value' --output text --with-decryption`
+SFTP_PRIVATE_KEY=`aws --region eu-west-2 ssm get-parameter --name mock_ftp_sftp_server_SFTPuser_private_key --query 'Parameter.Value' --output text --with-decryption`
+PUBLIC_IP=`aws --region eu-west-2 ssm get-parameter --name mock_FTP_SFTP_Server_Public_IP --query 'Parameter.Value' --output text --with-decryption`
```

Comment:
pubic IP? you might want to consider a better variable name than this.
also what are you doing with these variables, you're not setting them anywhere and you're not doing anything with them after fetching?